packages/docusaurus-theme-redoc/src/types/common.ts

```diff
@@ -1,17 +1,45 @@
 import type { Props as LayoutProps } from '@theme/Layout';
-import type { ObjectDescriptionProps } from 'redoc';
+import type { ObjectDescriptionProps, RedocRawOptions } from 'redoc';
 import type { OpenAPISpec } from 'redoc/typings/types';
 
 export type ParsedSpec = OpenAPISpec;
 
 export interface SpecProps {
+  /**
+   * Spec to use, already loaded previously
+   */
   spec: ParsedSpec;
-  url?: string;
-  isSpecFile?: boolean;
+  /**
+   * When spec not provided, load the spec from docusaurus config
+   * fallback to first configuration if not provided
+   */
+  id?: string;
+  /**
+   * docusaurus theme to use
+   */
   themeId?: string;
 }
 
-export type RedocProps = SpecProps;
+export type SpecPropsWithUrl = Omit<SpecProps, 'id'> & {
+  /**
+   * Absolute path to the spec file used
+   */
+  url?: string;
+};
+
+export type RedocProps = SpecProps & {
+  /**
+   * FIXME - incorrect name - should be isExternalUrl
+   */
+  isSpecFile?: boolean;
+  className?: string;
+  optionsOverrides?: RedocRawOptions;
+  /**
+   * External URL to load spec file from
+   * FIXME - incorrect name - should be externalUrl
+   */
+  url?: string;
+};
```

Comment:
>I think isSpecFile can be removed now since I've added a tester that check if the user provided an url starting with http

It could also be a relative url to a yaml file in the static directory.

> When you provide url to Redoc, it means you want to render an external swagger link. It should be renamed... (or at least documented)

Currently it works as the download url as the processed one removes some formatting which makes it hard to read 

The external url is not intended, will see if the 2 usecases can be separated

Reply:
Feel free to commit over my PR or tell me what to do.

This is weak part of this MR

Reply:
> It could also be a relative url to a yaml file in the static directory.

Yes, I've made a function that computes if the file is inside the doc or on another website. That's just checking if the URL start with http, no need to ask something as a config attribute